js/src/common/utils/ItemList.ts

```diff
@@ -13,65 +22,141 @@ class Item {
+    return Object.keys(this._items).length === 0;
   }
 
   /**
    * Check whether an item is present in the list.
    */
   has(key: string): boolean {
-    return !!this.items[key];
+    return Object.keys(this._items).includes(key);
   }
 
   /**
    * Get the content of an item.
    */
-  get(key: string): any {
-    return this.items[key].content;
+  get(key: string): T {
+    return this._items[key].content;
   }
 
   /**
    * Add an item to the list.
    *
    * @param key A unique key for the item.
    * @param content The item's content.
-   * @param [priority] The priority of the item. Items with a higher
-   *     priority will be positioned before items with a lower priority.
+   * @param priority The priority of the item. Items with a higher priority
+   * will be positioned before items with a lower priority.
    */
-  add(key: string, content: any, priority: number = 0): this {
-    this.items[key] = new Item(content, priority);
+  add(key: string, content: T, priority: number = 0): this {
+    this._items[key] = new Item(content, priority);
 
     return this;
   }
 
+  // TODO: [Flarum 2.0] Remove deprecated `.replace()` syntax.
+
+  //? This is the deprecated syntax.
   /**
-   * Replace an item in the list, only if it is already present.
+   * Replace an item and/or priority in the list, only if it is already present.
+   *
+   * If `content` or `priority` are `null`, these values will not be replaced.
+   *
+   * If the provided `key` is not present, nothing will happen.
+   *
+   * @deprecated Please use the `changePriority` method to set the priority for
+   * an item. This will be removed in Flarum 2.0.
+   *
+   * @param key The key of the item in the list
+   * @param content The item's new content
+   * @param priority The item's new priority
+   *
+   * @example <caption>Replace priority and not content.</caption>
+   * items.replace('myItem', null, 10);
+   *
+   * @example <caption>Replace content and not priority.</caption>
+   * items.replace('myItem', <p>My new value.</p>);
+   *
+   * @example <caption>Replace content and priority.</caption>
+   * items.replace('myItem', <p>My new value.</p>, 10);
    */
-  replace(key: string, content: any = null, priority: number = null): this {
-    if (this.items[key]) {
-      if (content !== null) {
-        this.items[key].content = content;
-      }
+  replace(key: string, content?: T | null, priority?: number): this;
 
-      if (priority !== null) {
-        this.items[key].priority = priority;
-      }
+  //? This is the new syntax.
+  /**
+   * Replaces an item's content, if the provided item key exists.
+   *
+   * If the provided `key` is not present, nothing will happen.
+   *
+   * @param key The key of the item in the list
+   * @param content The item's new content
+   *
+   * @example <caption>Replace item content.</caption>
+   * items.replace('myItem', <p>My new value.</p>);
+   *
+   * @example <caption>Replace item content and priority.</caption>
```

Comment:
This example doesnt seem to apply here

Reply:
It was meant to illustrate the chaining and show an example of how to use the new syntax instead of the deprecated version.